### PR TITLE
feat: stack arrays (test data in workspace stack_array_stuff)

### PR DESCRIPTION
**Pull request by yellowsubmarine1447**
Arrays on the stack. Stack arrays.

### CODE REVIEW

**Comment by dqna64** [comment]
verify:

Screenshot 2024-06-13 at 18 26 25

Screenshot 2024-06-13 at 18 27 42

LGTM 👍 nice work guys :DD

**Comment by dqna64** [comment]
In integer reconstruction code, need to account for:

- endianness of system
- what data type is being reconstructed `(signed | unsigned) (short | int | float | double)` etc (signed values require two's complement in reconstruction code)

File: debugger/src/gdb_scripts/custom_next.py [hunk 1]
```diff
@@ -501,10 +502,13 @@ def split_gdb_examine(gdb_examine_data, cellSize):
         print(f"{line=}")
         for number in line.split(":", 1)[1].split():
             print(f"{number=}")
+            # gdb returns each number byte as a signed value (i.e. in the range [-128, 127] instead of [0, 255])
+            # negative numbers break when byte values are bit shifted and added together
+            number = int(number) % 0x100
```
Comment (dqna64) [inline debugger/src/gdb_scripts/custom_next.py:507]: use `x/4ub addr` instead to get bytes in unsigned decimal format eg

```
(gdb) x/4b arr
0xaaaae40016b0:	-125	15	-52	-127

(gdb) x/4ub arr
0xaaaae40016b0:	131	15	204	129
```

**Review by dqna64 — CHANGES_REQUESTED** [review]

**Comment by Shubh141** [comment]
lgtm

**Review by Shubh141 — APPROVED** [review]

**Review by Shubh141 — APPROVED** [review]

**Review by Shubh141 — APPROVED** [review]

**Review by Shubh141 — APPROVED** [review]

**Review by Shubh141 — APPROVED** [review]

**Pull request merged**